src/Compilers/CSharp/Portable/Symbols/Source/SourceOrdinaryMethodOrUserDefinedOperatorSymbol.cs

```diff
@@ -0,0 +1,275 @@
+                                                                  out _lazyParameters, alsoCopyParamsModifier: false);
+                    this.FindExplicitlyImplementedMemberVerification(overriddenOrExplicitlyImplementedMethod, diagnostics);
+                    TypeSymbol.CheckNullableReferenceTypeMismatchOnImplementingMember(this.ContainingType, this, overriddenOrExplicitlyImplementedMethod, isExplicit: true, diagnostics);
+                }
+                else
+                {
+                    Debug.Assert(_lazyExplicitInterfaceImplementations.IsDefault);
+                    _lazyExplicitInterfaceImplementations = ImmutableArray<MethodSymbol>.Empty;
+
+                    Debug.Assert(_lazyReturnType.CustomModifiers.IsEmpty);
+                }
+            }
+
+            return overriddenOrExplicitlyImplementedMethod;
+        }
+
+        protected abstract void ExtensionMethodChecks(BindingDiagnosticBag diagnostics);
+
+        protected abstract MethodSymbol? FindExplicitlyImplementedMethod(BindingDiagnosticBag diagnostics);
+
+        protected abstract TypeSymbol? ExplicitInterfaceType { get; }
+
+        internal sealed override int ParameterCount
+        {
+            get
+            {
+                if (!_lazyParameters.IsDefault)
+                {
+                    int result = _lazyParameters.Length;
+                    Debug.Assert(result == GetParameterCountFromSyntax());
+                    return result;
+                }
+
+                return GetParameterCountFromSyntax();
+            }
+        }
+
+        protected abstract int GetParameterCountFromSyntax();
+
+        public sealed override ImmutableArray<ParameterSymbol> Parameters
+        {
+            get
+            {
+                LazyMethodChecks();
+                return _lazyParameters;
+            }
+        }
+
+        public sealed override TypeWithAnnotations ReturnTypeWithAnnotations
+        {
+            get
+            {
+                LazyMethodChecks();
+                return _lazyReturnType;
+            }
+        }
+
+        internal sealed override bool IsExplicitInterfaceImplementation
+        {
+            get
+            {
+                return MethodKind == MethodKind.ExplicitInterfaceImplementation;
+            }
+        }
+
+        public sealed override ImmutableArray<MethodSymbol> ExplicitInterfaceImplementations
+        {
+            get
+            {
+                LazyMethodChecks();
+                return _lazyExplicitInterfaceImplementations;
+            }
+        }
+
+        public sealed override ImmutableArray<CustomModifier> RefCustomModifiers
+        {
+            get
+            {
+                LazyMethodChecks();
+                return _lazyRefCustomModifiers;
+            }
+        }
+
+        internal override void AfterAddingTypeMembersChecks(ConversionsBase conversions, BindingDiagnosticBag diagnostics)
```

Comment:
Should this be sealed? It doesn't appear to be overridden by any derived types today.

Reply:
> Should this be sealed? It doesn't appear to be overridden by any derived types today.

No, sealing a method is supposed to communicate intent to prevent derived types from changing implementation. I do not have this intent here.